ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -157,13 +157,15 @@ public KsqlConfig(Map<?, ?> props) {
         StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, defaultCacheMaxBytesBufferingConfig);
     ksqlStreamConfigProps.put(StreamsConfig.NUM_STREAM_THREADS_CONFIG, defaultNumberOfStreamsThreads);
 
-    for (Object propKey: props.keySet()) {
-      if (propKey.toString().toLowerCase().startsWith(KSQL_CONFIG_PREPERTY_PREFIX)) {
-        ksqlConfigProps.put(propKey.toString(), props.get(propKey));
+    for (Map.Entry<?, ?> entry : props.entrySet()) {
+      final String key = entry.getKey().toString();
+      if (key.toLowerCase().startsWith(KSQL_CONFIG_PREPERTY_PREFIX)) {
+        ksqlConfigProps.put(key, entry.getValue());
       } else {
-        ksqlStreamConfigProps.put(propKey.toString(), props.get(propKey));
```

Comment:
Hmm. What's the findbugs suggestion here? Possible NPE?

Reply:
It is the `for` loop. performance optimization to iterate over the `entrySet` rather than  `keySet` and `get`